build/nuget-release.yml

```diff
@@ -0,0 +1,123 @@
+
+  - stage: UnitTests
+    displayName: Unit Tests
+    dependsOn: Build
+    condition: succeeded()
+    jobs:
+      - job: UnitTests
+        displayName: 'Run unit tests'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - template: test/run-unit-tests.yml@templates
+            parameters:
+              dotnetSdkVersion: '$(DotNet.Sdk.Version)'
+              projectName: '$(Project).Tests.Unit'
+
+  - stage: IntegrationTests
+    displayName: Integration Tests
+    dependsOn: Build
+    condition: succeeded()
+    jobs:
+      - job: IntegrationTests
+        displayName: 'Run integration tests'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - template: templates/run-service-bus-test-worker.yml
+            parameters:
+              buildVersion: $(Build.BuildId)
+              healthPort: $(Arcus_Health_Port)
+          - template: test/run-integration-tests.yml@templates
+            parameters:
+              dotnetSdkVersion: '$(DotNet.Sdk.Version)'
+              projectName: '$(Project).Tests.Integration'
+
+  - stage: Release
+    displayName: 'Release to NuGet.org'
+    dependsOn: IntegrationTests
+    condition: succeeded()
+    jobs:
+      - job: PushToNuGet
+        displayName: 'Push to NuGet.org'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - template: github/create-release.yml@templates
+            parameters:
+              repositoryName: 'arcus-azure/arcus.messaging'
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.Messaging.Health/$(Build.BuildNumber))
+                ```shell
+                PM > Install-Package Arcus.Messaging.Health --Version $(Build.BuildNumber)
+                ```
+            condition: not(contains(variables['Package.Version'], '-'))
+          - template: create-pre-release.yml@templates
+            parameters:
+              repositoryName: 'arcus-azure/arcus.messaging'
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.Messaging.Health/$(Build.BuildNumber))
+                ```shell
+                PM > Install-Package Arcus.Messaging.Health --Version $(Build.BuildNumber)
+                ```
+            condition: contains(variables['Build.BuildNumber'], '-')
```

Comment:
This is on purpose as the current approach does not work as intended. See https://dev.azure.com/codit/Arcus/_build/results?buildId=39734&view=logs&s=e2660204-6006-59e6-789e-83c238ff80a0

I've tested this approach in a seperate branch and it works fine, I think the reason why it was not working is because it does the decision when it creates the pipeline or so.

Reply:
Aha, OK. Then my comments in the other PR's are also obsolete.